src/main/java/org/kairosdb/core/queue/MemoryQueueProcessor.java

```diff
@@ -40,16 +41,17 @@ public MemoryQueueProcessor(
 
     @Override
     public void addReportedMetrics(final PeriodicMetrics periodicMetrics) {
+        periodicMetrics.recordGauge("queue/persisted_samples", m_persistedSamplesCount.getAndSet(0L));
+        periodicMetrics.recordGauge("queue/dropped_samples", m_droppedSamplesCount.getAndSet(0L));
         periodicMetrics.recordGauge("queue/process_count", m_readFromQueueCount.getAndSet(0));
-        periodicMetrics.recordGauge("queue/memory_queue.size", m_queue.size());
+        periodicMetrics.recordGauge("queue/memory_queue_size", m_queue.size());
     }
 
     @Override
     public void put(final DataPointEvent dataPointEvent) {
-        try {
-            m_queue.put(dataPointEvent);
-        } catch (final InterruptedException e) {
-            logger.error("Error putting data", e);
+        if (!m_queue.offer(dataPointEvent)) {
+            m_droppedSamplesCount.addAndGet(dataPointEvent.getDataPoint().getSampleCount());
+            logger.error("Error putting data");
```

Comment:
nit: Can you add that the queue is full to this log because the the docs say that would be the only reason offer should fail.

Reply:
Done!